test/example-types/facet.js

```diff
@@ -434,5 +435,82 @@ describe('facet', () => {
         ],
       })
     })
+
+    describe('should always include checked values', () => {
+      let Data = [
+        { _id: 1, name: '1' },
+        { _id: 2, name: '2' },
+        { _id: 3, name: '3' },
+        { _id: 4, name: '4' },
+      ]
+      let mongoIdData = [
+        { _id: '5e9dbd76e991760021124966', name: 'Automation' },
+        { _id: '5cde2658dc766b0030c67dae', name: 'Fowlkes (MO)' },
+        { _id: '5d1ca49436e1d20038f8c84f', name: 'Customer Experience' },
+        { _id: '5ce30b403aa154002d01b9ed', name: 'Government Division' },
+      ]
+      let node = {
+        key: 'id',
+        field: '_id',
+        type: 'facet',
+        label: {
+          collection: null,
+          foreignField: '_id',
+          fields: ['name'],
+        },
+        mode: 'include',
+        optionsFilter: '',
+        size: 2,
+      }
+      it('when there are missed values', async () => {
+        ;(node.label.collection = Data), (node.values = [4])
+
+        let result = await facet.result(node, agg => mingo.aggregate(Data, agg))
+        let ids = _.map(({ name }) => _.toString(name), result.options)
+        expect(result.options.length).to.equal(2)
+        expect(_.includes('4', ids)).to.be.true
+      })
+      it('when there is no missed value', async () => {
+        ;(node.label.collection = Data), (node.values = [1])
+        let result = await facet.result(node, agg => mingo.aggregate(Data, agg))
+        let ids = _.map(({ name }) => _.toString(name), result.options)
+        expect(result.options.length).to.equal(2)
+        expect(_.includes('1', ids)).to.be.true
+      })
+      it('when there are missed values and isMongoId: true', async () => {
+        let collection = _.map(
+          ({ _id, name }) => ({ _id: ObjectID(_id), name }),
+          mongoIdData
+        )
+        node.isMongoId = true
+        ;(node.label.collection = collection),
+          (node.values = ['5ce30b403aa154002d01b9ed'])
+
+        let result = await facet.result(node, agg =>
+          mingo.aggregate(collection, agg)
+        )
+        let ids = _.map(({ name }) => _.toString(name), result.options)
+
+        expect(result.options.length).to.equal(2)
+        expect(_.includes('5ce30b403aa154002d01b9ed', ids)).to.be.true
+      })
+      it('when there is no missed value and isMongoId: true', async () => {
```

Comment:
this test is probably overkill and could be removed in my opinion.

Reply:
I think we have to include all conditions. 
 1,do not has prop isMongoId(a, missing values during search, b do not missing value during search). 2.isMongoId: true (a, missing values during search, b do not missing value during search)